sklearn/utils/estimator_checks.py

```diff
@@ -2565,6 +2567,60 @@ def check_classifiers_regression_target(name, estimator_orig):
         assert_raises_regex(ValueError, msg, e.fit, X, y)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_estimator_sparse_dense(name, estimator_orig):
+    rng = np.random.RandomState(52)
+    estimator = clone(estimator_orig)
+    estimator_sp = clone(estimator_orig)
+    tags = _safe_tags(estimator_orig)
+    centers = 2 if tags["binary_only"] else None
+    X, y = make_blobs(random_state=rng, cluster_std=0.5, centers=centers)
+    # for put some points to zero to have a little bit of sparsity
+    # TODO: update this with enforce_estimator_tags_X when #14705 is merged
+    if tags["requires_positive_X"]:
+        X -= np.min(X) - 1.
+    X_csr = sparse.csr_matrix(X)
+    if tags["multioutput"] or tags["multioutput_only"]:
+        y = y[:, np.newaxis]
+
+    for sparse_format in ['csr', 'csc', 'dok', 'lil', 'coo', 'dia', 'bsr']:
+        X_sp = X_csr.asformat(sparse_format)
+        if name in ['Scaler', 'StandardScaler']:
+            estimator.set_params(with_mean=False)
```

Comment:
I agree, I'll remove them here
I checked and it seems that these variants are in fact actually already tested here `sklearn.preprocessing.tests.test_scaler_without_centering` (by the way I didn't find any `Scaler` class in scikit-learn so I guess it was just really about `StandardScaler`), so I guess I don't need to implement a new test ?

Reply:
This comment still needs to be addressed.

> (by the way I didn't find any Scaler class in scikit-learn so I guess it was just really about StandardScaler), 

And `RobustScaler`.

> so I guess I don't need to implement a new test ?

If `RobustScaler` is also tested, then no need for a new test. I haven't checked.

Reply:
Yes, `RobustScaler`is tested too here: https://github.com/scikit-learn/scikit-learn/blob/1c116807ce1e925f73c87c34e586520d455682d6/sklearn/preprocessing/tests/test_data.py#L1123

So I removed the lines with the "ifs" here in `estimator_checks`, and added in `StandardScaler` and `RobustScaler` the `xfail` estimator tag for this `check_estimator_dense_sparse` test (I explain in the `xfail` message that I exclude this test because it's not possible for the default parameters, but the test is made in `test_scaler_without_centering/test_robust_scaler_equivalence_dense_sparse`) (see commit https://github.com/scikit-learn/scikit-learn/pull/13246/commits/e32e6aa87c51f47fd50d62536bb9287d7966c8c5)


PS: I noticed however that `StandardScaler`'s test is more intensive (tests `transform`, `inverse_transform`, etc, when the one for `RobustScaler` only tests the `mean_` attribute) https://github.com/scikit-learn/scikit-learn/blob/1c116807ce1e925f73c87c34e586520d455682d6/sklearn/preprocessing/tests/test_data.py#L785
I thought I could make `RobustScaler`'s test as robust as `StandardScaler`'s one, but I don't know if it's the purpose of this PR maybe it just adds more to review ?